feature-banners-impl/src/main/java/io/novafoundation/nova/feature_banners_impl/presentation/banner/PromotionBannersMixin.kt

```diff
@@ -0,0 +1,108 @@
+package io.novafoundation.nova.feature_banners_impl.presentation.banner
+
+import android.content.Context
+import android.content.Intent
+import android.content.Intent.FLAG_ACTIVITY_NEW_TASK
+import android.graphics.drawable.Drawable
+import android.net.Uri
+import android.util.Log
+import coil.ImageLoader
+import coil.request.ImageRequest
+import io.novafoundation.nova.common.domain.ExtendedLoadingState
+import io.novafoundation.nova.common.domain.asError
+import io.novafoundation.nova.common.domain.asLoaded
+import io.novafoundation.nova.common.utils.LOG_TAG
+import io.novafoundation.nova.feature_banners_api.domain.PromotionBanner
+import io.novafoundation.nova.common.utils.asyncWithContext
+import io.novafoundation.nova.common.utils.toMap
+import io.novafoundation.nova.common.utils.withSafeLoading
+import io.novafoundation.nova.feature_banners_api.presentation.BannerPageModel
+import io.novafoundation.nova.feature_banners_api.presentation.ClipableImage
+import io.novafoundation.nova.feature_banners_api.presentation.PromotionBannersMixin
+import io.novafoundation.nova.feature_banners_api.presentation.PromotionBannersMixinFactory
+import io.novafoundation.nova.feature_banners_api.presentation.source.BannersSource
+import io.novafoundation.nova.feature_banners_impl.domain.PromotionBannersInteractor
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.launch
+
+class RealPromotionBannersMixinFactory(
+    private val imageLoader: ImageLoader,
+    private val context: Context,
+    private val promotionBannersInteractor: PromotionBannersInteractor
+) : PromotionBannersMixinFactory {
+
+    override fun create(source: BannersSource): PromotionBannersMixin {
+        return RealPromotionBannersMixin(
+            promotionBannersInteractor,
+            imageLoader,
+            context,
+            source
+        )
+    }
+}
+
+class RealPromotionBannersMixin(
+    private val promotionBannersInteractor: PromotionBannersInteractor,
+    private val imageLoader: ImageLoader,
+    private val context: Context,
+    private val bannersSource: BannersSource
+) : PromotionBannersMixin {
+
+    override val bannersFlow = bannersSource.observeBanners()
+        .map { banners ->
+            val resources = loadResources(banners)
+            banners.map { mapBanner(it, resources) }
+        }.withSafeLoading()
```

Comment:
I suggest we always do `share/shareInBackground` in a mixin. We can do that by passing `CoroutineContext` to the mixin factory and then to the mixin itself so mixin is a fully-featured component similar to ViewModel that does not require any extra actions on top of just creating it and wiring with UI